### PR TITLE
Refactor GetStorer to return error

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- add error on return of `GetStorer` method
  
## Proposed Changes
- add error on return of `GetStorer` method
- remove all implementations of `ChainStorerMock` and `ChainStorerStub` and only use the ones from `testscommon`

## Testing procedure
- system test

### CODE REVIEW

File: process/sync/storageBootstrap/metaStorageBootstrapper.go [hunk 1]
```diff
@@ -122,7 +125,13 @@ func (msb *metaStorageBootstrapper) cleanupNotarizedStorage(metaBlockHash []byte
 			"hash", shardHeaderHash)
 
 		hdrNonceHashDataUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(shardHeader.GetShardID())
-		storer := msb.store.GetStorer(hdrNonceHashDataUnit)
+		storer, err := msb.store.GetStorer(hdrNonceHashDataUnit)
+		if err != nil {
+			log.Debug("could not get storage unit",
+				"unit", hdrNonceHashDataUnit,
+				"error", err.Error())
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply process/sync/storageBootstrap/metaStorageBootstrapper.go:132]: correct, added return

File: dataRetriever/interface.go [hunk 2]
```diff
@@ -39,10 +39,32 @@ func (ut UnitType) String() string {
 		return "BootstrapUnit"
 	case StatusMetricsUnit:
 		return "StatusMetricsUnit"
+	case TxLogsUnit:
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/interface.go:42]: done

File: dataRetriever/errors.go [hunk 3]
```diff
@@ -244,3 +244,6 @@ var ErrInvalidNumOfPeerAuthentication = errors.New("invalid num of peer authenti
 
 // ErrNilPeerShardMapper signals that a nil peer shard mapper has been provided
 var ErrNilPeerShardMapper = errors.New("nil peer shard mapper")
+
+// ErrKeyNotFound signals that the provided key was not found
+var ErrKeyNotFound = errors.New("key not found")
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply dataRetriever/errors.go:249]: replaced with `ErrStorerNotFound`

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by popenta — APPROVED** [review]
@@                            Log scanner                                  @@
##                            refactor_getstorer                           ##
================================================================================
- Known Warnings              24
- New Warnings                12
- Known Errors                0
- New Errors                  2
- Panics                      0
================================================================================

**Review by popenta — APPROVED** [review]
System test passed

**Review by ssd04 — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**